app/jobs/analyze_change_job.rb

```diff
@@ -73,39 +73,44 @@ def perform(to_version, from_version = nil, compare_earliest = true)
   # This may shortly become a simple call to some processing server endpoint:
   # https://github.com/edgi-govdata-archiving/web-monitoring-db/issues/404
   def analyze_change(change)
+    text_diff_promise = Concurrent::Promise.execute { Differ.for_type!('html_text_dmp').diff(change)['diff'] }
+    source_diff_promise = Concurrent::Promise.execute { Differ.for_type!('html_source_dmp').diff(change)['diff'] }
+    links_diff_promise = Concurrent::Promise.execute { Differ.for_type!('links_json').diff(change)['diff'] }
```

Comment:
The docs say `Concurrent::Promise` has been deprecated and replaced with `Concurrent::Promises` (singular vs. plural for maximum hard-to-spot-ness). The new module has `#future` instead of `#execute`, but it looks like you’d otherwise use it the same here.

Since we create these and resolve them in the same scope and before touching `change` again, it’s probably not important, but maybe more safety-conscious to pass `change` as an argument.

Also maybe worth making a method for this like you did `diff_changes`?

```rb
schedule_diff(type, change)
  Concurrent::Promises.future(type, change) { |t, c| Differ.for_type!(t).diff(change)['diff'] }
end

Reply:
Thanks for digging into the `Concurrent::Promise` docs and pointing me in the right direction. 👍